src/components/Territory.vue

```diff
@@ -108,20 +109,29 @@ export default {
       checkinTerritory: 'territory/checkinTerritory',
     }),
 
-    async checkIn() {
-      if (window.confirm('Ready to check-in the territory?')) {
-        this.isLoading = true;
-        await this.resetNHRecords(this.id);
-        await this.getTerritory(this.id);
-        await this.checkinTerritory({
-          territoryId: this.id,
-          userId: get(this.territory, 'status.publisher.id'),
-          username: this.user.username,
-        });
-        this.isLoading = false;
-        await this.$router.push({ name: 'home' });
-        this.checkInToast('success');
-      }
+    checkIn() {
+      this.$bvModal.msgBoxConfirm('Ready to check-in the territory?', {
```

Comment:
almost there... can you follow the async/await pattern for promises instead of using `[promise].then(...)`?